core/dbt/cli/flags.py

```diff
@@ -49,13 +49,33 @@ def convert_config(config_name, config_value):
     # This function should take care of converting the values from config and original
     # set_from_args to the correct type
     ret = config_value
-    if config_name.lower() == "warn_error_options":
+    if config_name.lower() == "warn_error_options" and type(config_value) == dict:
```

Comment:
Why is this change necessary? Shouldn't warn_error_options always be a dictionary?

Docs: https://docs.getdbt.com/reference/global-configs#warnings-as-errors

Reply:
There are other places that passes in something that's not a dict. I can do another round of verify it is a dict and update input from other places.

Reply:
I remember why, it is because the default value for `warn_error_options` in [UserConfig](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/project.py#L245) is None, so if you pass in a UserConfig object the original logic would fail

Reply:
👍 gotcha. `config_value is not None` would be more precise here then